content/blog/2025-02-02-datafusion-ballista-43.0.0.md

```diff
@@ -0,0 +1,128 @@
+
+[delta-rs]: https://github.com/delta-io/delta-rs
+
+The most significant enhancement in this release is the deprecation of `BallistaContext`, which has been superseded by the DataFusion `SessionContext`. This change enables DataFusion applications written in Rust to execute on a Ballista cluster with minimal modifications. Beyond simplifying migration and reducing maintenance overhead, this update introduces distributed write functionality to Ballista for the first time, significantly enhancing its capabilities. At the moment there is a gap between DataFusion and Ballista, which we will try to bridge in the future.
+
+```rust
+use ballista::prelude::*;
+use datafusion::prelude::*;
+
+#[tokio::main]
+async fn main() -> datafusion::error::Result<()> {
+
+  // Instead of creating classic SessionContext
+  // let ctx = SessionContext::new();
+  
+  // create DataFusion SessionContext with ballista standalone cluster started
+  // let ctx = SessionContext::standalone().await;
+
+  // create DataFusion SessionContext with ballista remote cluster started
+  let ctx = SessionContext::remote("df://localhost:50050").await;
+
+  // register the table
+  ctx.register_csv("example", "tests/data/example.csv", CsvReadOptions::new()).await?;
+
+  // create a plan to run a SQL query
+  let df = ctx.sql("SELECT a, MIN(b) FROM example WHERE a <= b GROUP BY a LIMIT 100").await?;
+
+  // execute and print results
+  df.show().await?;
+  Ok(())
+}
+```
+
+Additionally, Ballista’s versioning scheme has been aligned with that of DataFusion, ensuring that Ballista's version number reflects the compatible DataFusion version..
+
+### Removal of Experimental Features
+
+Ballista had grown in scope to include several experimental features in various states of completeness. Some features have been removed from this release in an effort to strip Ballista back to its core and make it easier to maintain and extend.
+
+Specifically, the caching subsystem, object store registry, plugin subsystem, key-value stores for persistent scheduler state, and the UI have been removed.
+
+### Performance & Scalability
+
+Ballista has significantly leveraged the advancements made in the DataFusion project over the past year. Benchmark results demonstrate notable improvements in performance, highlighting the impact of these enhancements:
+
+Per query comparison:
+
+<img
+src="https://github.com/apache/datafusion-ballista/raw/e9e8f9a783d9dbc8b40f3e57e2a7294788df6b93/docs/source/_static/images/tpch_queries_compare.png"
```

Comment:
I wonder if it would be better to add these images as part of this PR rather than link to these GitHub URLs?